store/sqlstore/post_store.go

```diff
@@ -119,10 +119,10 @@ func (s *SqlPostStore) Save(post *model.Post) store.StoreChannel {
 				post.Type != model.POST_JOIN_TEAM && post.Type != model.POST_LEAVE_TEAM &&
 				post.Type != model.POST_ADD_TO_CHANNEL && post.Type != model.POST_REMOVE_FROM_CHANNEL &&
 				post.Type != model.POST_ADD_TO_TEAM && post.Type != model.POST_REMOVE_FROM_TEAM {
-				s.GetMaster().Exec("UPDATE Channels SET LastPostAt = :LastPostAt, TotalMsgCount = TotalMsgCount + 1 WHERE Id = :ChannelId", map[string]interface{}{"LastPostAt": time, "ChannelId": post.ChannelId})
+				s.GetMaster().Exec("UPDATE Channels SET LastPostAt = (CASE WHEN :LastPostAt > LastPostAt THEN :LastPostAt ELSE LastPostAt END), TotalMsgCount = TotalMsgCount + 1 WHERE Id = :ChannelId", map[string]interface{}{"LastPostAt": time, "ChannelId": post.ChannelId})
```

Comment:
The changes in this file are the actual solution to the problem, update the LastPostAt only if the saved post is older than the current LastPostAt. That fixes the problem with loading a dump and get the Channels LastPostAt date out of sync with the posts.

Reply:
nit: use `GREATEST(:LastPostAt, LastPostAt)` for simplicity?